### PR TITLE
Big push for ability localization (sorry)

### PR DESCRIPTION
**Pull request by gdirk07**
1. removed Tachyons (was used as some quick way to write CSS when I was still learning) from the project and updated SearchBar.tsx
2. Fix an error where there was a white bar at top of the app (overflow in APP.css)
3. Added new value to Ability "localizedName" which we can build off of for multi language support
4. Cleaned up App.tsx to remove unneeded states, and functions, will help with future optimizations on renders since most of the clutter is gone
5. fetchAbilities moved outside into its own function to allow async solution
6. Made it so abilities now render the localized name instead of the generic name from the API

The reason this commit was long was because I was struggling to have the localizedName load because of the new hooks in PokemonDisplay, this caused me searching for a solution in App.tsx which got me cleaning it up. The solution now is before the pokemon renders we fetch the mon's abilities. I don't like this solution because now pokemon load slowly while it waits for up to 3 fetches.

Alternative solution would be a blank placeholder until the promise resolves and then re-render the ability display. I wasn't able to get far with it because of how the flow exists. First we fetch pokemon (async) -> create pokemon -> create abilities (async)-> display, while it did work, I couldn't get the hooks to re render the object when the promise resolved for abilities (get useEffect to detect that pokemonObject.abilities changed for example) , except for one time but then it continued to render loop forever. This is something we should look at as it will also effect future components like moves.

### CODE REVIEW

File: pokemon-checker/src/factories/PokemonFactory.ts [hunk 1]
```diff
@@ -49,14 +49,19 @@ export class PokemonFactory {
...
     return createdPokemon;
   };
 
+  public fetchAbilities = async (pokemon : PokemonDTO): Promise<PokemonDTO> => {
+    await Promise.all(pokemon.abilities.map(async (ability) => {
+      if (!ability[0].hasFullData)
+        //if retrieved from repository it likely has the data, otherwise...
+        await this.abilityService.getFullAbilityDef(ability[0]);
+    }));
+    console.log("complete");
```
Comment (jeremy-jtlo) [inline pokemon-checker/src/factories/PokemonFactory.ts:61]: Did we mean to leave a `console.log` here?

Comment (gdirk07) [reply pokemon-checker/src/factories/PokemonFactory.ts:61]: leftover from when I was debugging we can remove it

File: pokemon-checker/src/components/PokemonDisplay/PokemonDisplay.tsx [hunk 2]
```diff
@@ -27,33 +26,26 @@ export const PokemonDisplay = (props: displayProps) => {
...
   const createPokemonObject = useCallback(
-    (pokemonRetrieved: IPokemonData): void => {
-      let pokemonToDisplay = pokemonFactory.createPokemon(pokemonRetrieved);
-      setPokemonObject(pokemonToDisplay);
+    (url: string) => {
+      if (url && url.length > 0) {
+        getPokemonData(url)
+          .then((data) => pokemonFactory.createPokemon(data))
+          .then((pokemon) => pokemonFactory.fetchAbilities(pokemon))
+          .then((pokemon) => setPokemonObject(pokemon));
```
Comment (jeremy-jtlo) [inline pokemon-checker/src/components/PokemonDisplay/PokemonDisplay.tsx:40]: Hot dang this is what I was trying to do in the service

Thanks for clearing that up, I was getting bodied by the `.then` calls after a successful fetch

**Comment by jeremy-jtlo** [comment]
Managed to run this locally after updating node on my laptop. No big issues and all unit tests pass. Good stuff.

**Pull request merged**